src/util/prolog/prologResult.ts

```diff
@@ -1,42 +1,37 @@
-import * as ld from "lodash";
-
 // First Group will be the variable value. Doesn't fetch the variable names
-const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+const variableAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+const booleanAnswerRegExp = /^(true|false)( ;|,|.)$/;
 
 export class PrologResult {
 	private readonly rawResults: string[];
-	private readonly results: Map<string, Map<number, string|boolean>>;
+	private results: Map<string, Array<string | boolean>> | undefined;
 
 	constructor(rawResults: string[]) {
 		this.rawResults = rawResults;
-		this.results = new Map();
 	}
 
-	public getResults(): Map<string, Map<number, string|boolean>> {
-		if (this.results.size !== 0) { return this.results; }
+	public getRawResults() {
+		return this.rawResults;
+	}
 
-		const results = new Map();
-		let i: number;
+	/**
+		* Generates (if needed) and returns the results.
+		*
+		*
+		*/
+	public getResults(): Map<string, Array<string | boolean>> {
```

Comment:
Nit: To protect the encapsulation of `PrologResult`, you should either change the return type of `getResults` to `ReadonlyMap<...>` or create a deep copy before returning!